streaming/rrdpush.c

```diff
@@ -581,12 +581,12 @@ static int rrdpush_sender_thread_connect_to_master(RRDHOST *host, int default_po
               , host->os
               , host->timezone
               , (host->tags) ? host->tags : ""
-              , (host->system_info->os_name) ? host->system_info->os_name : ""
-              , (host->system_info->os_id) ? host->system_info->os_id : ""
-              , (host->system_info->os_id_like) ? host->system_info->os_id_like : ""
-              , (host->system_info->os_version) ? host->system_info->os_version : ""
-              , (host->system_info->os_version_id) ? host->system_info->os_version_id : ""
-              , (host->system_info->os_detection) ? host->system_info->os_detection : ""
+              , (host->system_info->host_os_name) ? host->system_info->host_os_name : ""
+              , (host->system_info->host_os_id) ? host->system_info->host_os_id : ""
+              , (host->system_info->host_os_id_like) ? host->system_info->host_os_id_like : ""
+              , (host->system_info->host_os_version) ? host->system_info->host_os_version : ""
+              , (host->system_info->host_os_version_id) ? host->system_info->host_os_version_id : ""
+              , (host->system_info->host_os_detection) ? host->system_info->host_os_detection : ""
```

Comment:
Hi @amoss ,

After you review the code and I verified that everything was fixed, I began to test the PR, and I detected a problem with the stream. When I executed a request to `https://localhost:19999/host/darkstar/api/v1/info` on the master, I verified that the system identified correctly that I am running on Virtual Box

![err0](https://user-images.githubusercontent.com/49162938/72724746-16a33680-3b7c-11ea-8aa3-c39973acdd26.png)

on the other hand, I lost completely the OS description

![err1](https://user-images.githubusercontent.com/49162938/72724820-45211180-3b7c-11ea-9e6a-0af5ca5c69c6.png)

but when I call the `system-info.sh` direct from the terminal I can see the os information

![err3](https://user-images.githubusercontent.com/49162938/72724983-b19c1080-3b7c-11ea-8189-00bac4955375.png)

To confirm that this is a problem with this PR, I recompiled Netdata again on both sides(master and slave) using another branch and I could verify that the slave is streaming is working correctly.

Reply:
This looks quite serious - I'll setup some tests and check what has happened.

Reply:
I've had a look and in order to fix it I need to change the streaming protocol. The current streaming protocol is too closely connected to the internal structure of the `system_info` structure, and this structure has changed so that the field names communicated in the initial `STREAM` request no longer exist.

The `STREAM` request needs to communicate both the host and container versions of these fields, so that we can see the correct info in API requests and also in the GA results taht we deliver. What I propose is:

https://github.com/netdata/netdata/blob/6ba71e1704ef9fcd9335e1e4e1ad844b00a082fd/streaming/rrdpush.c#L558-L571'

* Change these to the HOST_ and CONTAINER_ versions.

https://github.com/netdata/netdata/blob/6ba71e1704ef9fcd9335e1e4e1ad844b00a082fd/streaming/rrdpush.c#L1398-L1401

* Add a check here that if we are failing on `SYSTEM_` create an error that the slave must be upgraded to the same version as the master for compatibility.

To some extent our hand are tied because we are communicating the names of internal field values in a protocol. I think it is reasonable to force an upgrade to the same version where necessary.

It's your AoR in the code - what would you like us to do?

Reply:
We can do the changes that you are suggesting, but we also need to keep in our mind backward compatibility. 
Case we are saying that is necessary to change the variable names, this will be a major change in the stream protocol, so it will be necessary to do something like I did on #7549 , as you can see there, we will need to do the following changes at `streaming/rrdpush.c`:

- Create a new funny message (This is the most important step);
- Change the protocol version (line 28 );
- Add a new if at line 650. Probably it is  a good time to move the lines `651 - 662` for a function
- We also need to change `rrdpush_receive` to accept a third kind of message, probably lines 1151-1158 are good candidate to become another function.
- Finally we need to do a last change inside the function `rrdpush_receiver_thread_spawn`, among the lines 1395 - 1402

Now we need to change the function `rrdhost_set_system_info_variable` inside `database/rrdhost.c`. The changes are

-  We need more one argument for the function that will identify the protocol version
- Depending of the version, we will use specific variable name to parse.

Of course this also brings the necessity to change `database/rrd.h`

Best regards!

Reply:
Created a new issue #7804 to track the work in updating the protocol.